include/votca/tools/structureparameters.h

```diff
@@ -0,0 +1,139 @@
+#include <unordered_map>
+
+namespace votca {
+namespace tools {
+
+/**
+ * \breif Supported and Standardized parameter types
+ **/
+enum StructureParameter {
+  CSG_Mass,
+  XTP_Mass,
+  CSG_Position,
+  XTP_Position,
+  MoleculeId,
+  ResidueId,
+  CSG_Charge,
+  XTP_Charge,
+  Element,
+  Symmetry,
+  ResidueType,
+  BeadId,
+  BeadType,
+  MoleculeType,
+  AtomContainerId,
+  AtomContainerType
+};
+
+/**
+ * \brief Provides a means to standardise the constructors of different classes
+ *
+ * The purpose of this class is for use primarily in io readers, it provides
+ * a means to standardize the use of templated classes in the readers.
+ *
+ * E.g. Say I have two atom classes
+ *
+ * class Atom1 {
+ *   public:
+ *     Atom1(StructureParameters parameters) :
+ *        id_(parameters.get<int>(StructureParameter::BeadId)),
+ *        mass_(parameters.get<double>(StructureParameter::Mass)){};
+ *
+ *   private:
+ *    int id_;
+ *    double mass_;
+ * };
+ *
+ * class Atom2 {
+ *   public:
+ *     Atom2(StructureParameters parameters) :
+ *        id_(parameters.get<int>(StructureParameter::BeadId)),
+ *        element_(parameters.get<string>(StructureParameter::Element)){};
+ *
+ *   private:
+ *    int id_;
+ *    string element_;
+ * };
+ *
+ * Pseudo code below, our file reader has a method that creates the atoms
+ * (shown below), now because we have a standardized constructor the templated
+ * file method can be used with either class without specialization.
+ *
+ * class FileReader {
+ *
+ *   template<class T>
+ *   T CreateAtomOrBead {
+ *
+ *     string element = "C";
+ *     int id = 1;
+ *     double mass = 12.01;
+ *
+ *     StructureParameters parameters;
+ *     parameters.set(StructureParameter::Element,element);
+ *     parameters.set(StructureParameter::Mass,mass);
+ *     parameters.set(StructureParameter::BeadId,id);
+ *
+ *     T atom_or_bead(parameters);
+ *     return atom_or_bead;
+ *   }
+ *
+ * };
+ **/
+class StructureParameters {
+
+ public:
+  void set(const StructureParameter parameter, boost::any value) noexcept {
+    assert(parameters_.count(parameter) == 0 &&
+           "Cannot set parameter it has already been set.");
+    parameters_[parameter] = value;
+  }
+
+  bool ParameterExist(StructureParameter parameter) const noexcept {
+    return parameters_.count(parameter);
+  }
+
+  template <class T>
+  T get(const StructureParameter parameter) {
+    if (parameters_.count(parameter) == 0) {
+      convertParameterIfPossible_(parameter);
+    }
+    assert(parameters_.count(parameter) &&
+           "StructureParameter is not stored in StructureParameters class");
+    assert(typeid(T) == parameters_.at(parameter).type() &&
+           "Cannot return boost any value from parameters class because it is "
+           "not being cast to the correct type");
+    return boost::any_cast<T>(parameters_.at(parameter));
+  }
+
+ private:
+  void convertParameterIfPossible_(const StructureParameter parameter);
```

Comment:
why no return type, where does it convert?

Reply:
why not a const reference?

Reply:
I am still unsure if I wan to keep this method. I thought it could be used to automatically convert a CSG parameters to XTP parameters if one is missing. For instance if I have the CSG_Position of an object but not the XTP_Position (Units are different), I could automatically convert the CSG_Position to the XTP_Position. The problem with this is that the class then needs to know about what units are used in xtp vs csg, what do you think?

Reply:
This method has been removed.